File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_05/java/Solution.java
package com.hi.dhl.algorithms.offer._05.java;

/**
 * <pre>
 *     author: dhl
 *     desc  :
 * </pre>
 */
class Solution {
    public String replaceSpace(String s) {
        StringBuffer buffer = new StringBuffer();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                buffer.append("%20");
            } else {
                buffer.append(s.charAt(i));
            }
        }
        return buffer.toString();
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_05/kotlin/Solution.kt
package com.hi.dhl.algorithms.offer._05.kotlin

/**
 * <pre>
 *     author: dhl
 *     desc  :
 * </pre>
 */

class Solution {
    fun replaceSpace(s: String): String {
        val buffer = StringBuffer()
        var i = 0
        val len = s.length
        while (i < len) {
            if (s[i] == ' ') {
                buffer.append("%20")
            } else {
                buffer.append(s[i])
            }
            i++
        }
        return buffer.toString()
    }
}